README.md

```diff
@@ -19,3 +18,44 @@ https://www.youtube.com/playlist?list=PLqnslRFeH2Upcrywf-u2etjdxxkL8nl7E
 - K-Means
 - AdaBoost
 - Linear Discriminant Analysis (LDA)
+
+## Installation and usage.
+
+This project has 2 dependencies.
+
+- `numpy` for the maths implementation and writing the algorithms
+- `Scikit-learn` for the data generation and testing.
+- `Matplotlib` for the plotting.
+- `Pandas` for loading data.
+
+You can install these using the command below!
+
+```sh
+# Linux or MacOS
+pip3 install -r requirements.txt
+
+# Windows
+pip install -r requirements.txt
+```
+
+You can run the files as following.
+
+```sh
+python -m algorithms.<algorithm-file>
```

Comment:
please keep the foldername "mlfromscratch". It's very common to use the same name for repo name and the folder with the code.